components/supportbee/sources/common/common.mjs

```diff
@@ -0,0 +1,67 @@
+import supportbee from "../../supportbee.app.mjs";
+
+export default {
+  props: {
+    supportbee,
+    db: "$.service.db",
+    timer: {
+      type: "$.interface.timer",
+      static: {
+        intervalSeconds: 15 * 60, // 15 minutes
+      },
+    },
+  },
+  methods: {
+    emitEvent(data) {
+      throw new Error("emitEvent is not implemented", data);
+    },
+    getResources({ ...args } = {}) {
+      throw new Error("getResources is not implemented", args);
+    },
+    _setLastResourceId(resourceId) {
+      this.db.set("lastResourceId", resourceId);
+    },
+    _getLastResourceId() {
+      return this.db.get("lastResourceId");
+    },
+  },
+  hooks: {
+    async deploy() {
+      const resources = await this.getResources({
+        params: {
+          per_page: 10,
+        },
+      });
+
+      resources.reverse().forEach(this.emitEvent);
+
+      this._setLastResourceId(resources[0].id);
+    },
+  },
+  async run() {
+    const lastResourceId = this._getLastResourceId();
+
+    let page = 1;
+
+    while (true) {
+      const resources = await this.getResources({
+        params: {
+          per_page: 100,
+          page,
+        },
+      });
+
+      resources.reverse().forEach(this.emitEvent);
+      this._setLastResourceId(resources[0].id);
```

Comment:
What happens if `resources` is empty `[]`? I think here you should check that right? something like
```suggestion
      if (resources.length) {
        this._setLastResourceId(resources[0].id);
      }
```

Reply:
Yes, you are right, sorry for this!